[Fong Yih Jie] iP

fongyj opened the pull request:
# DukePro
> "Your mind is for having ideas, not holding them." – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks. 
4. let it manage your tasks for you 🤨 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
- - -
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Duke duke = new Duke();
    }
}
```

Commit:
Add Gradle support

fongyj committed:
Add echo and bye functionality

fongyj committed:
Add add and list functionality

fongyj committed:
Add mark and unmark functionality

fongyj committed:
Add todo, event and deadline tasks

fongyj committed:
Add text ui testing

fongyj committed:
Add error handling

fongyj committed:
Add delete and more exception handling

fongyj committed:
Add save and load

fongyj committed:
Add date parsing

fongyj committed:
Merge branch 'branch-Level-7'

fongyj committed:
Merge branch 'branch-Level-8'

fongyj committed:
Refactor for more OOP

fongyj committed:
Organise classes into packages

fongyj committed:
Merge pull request #1 from nus-cs2103-AY2223S1/add-gradle-support

Add Gradle support

fongyj committed:
Add gradle support

fongyj committed:
Add JUnit tests

fongyj committed:
Add Jar file

fongyj committed:
Add JavaDoc

fongyj committed:
Refactor for coding standard

fongyj committed:
Add find feature

fongyj committed:
Merge branch 'branch-A-JavaDoc'

fongyj committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

fongyj committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

kelvinou01 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,62 @@
+    protected TaskList tasks;
+    protected Parser parser;
+
+    private boolean runningDuke;
```
Would a proposition be a clearer name for this boolean? (e.g. `dukeIsRunning` ?)

kelvinou01 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,139 @@
+     * @param verbose Boolean to indicate verbosity of Ui.
+     * @throws DukeException If command cannot be parsed or is invalid.
+     */
+    public void parse(String command, boolean verbose) throws DukeException {
```
Would a proposition be a clearer name for the boolean parameter? (e.g. `isVerbose`)?

kelvinou01 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,139 @@
+     * @throws DukeException If command cannot be parsed or is invalid.
+     */
+    public void parseTwo(String command) throws DukeException {
+        String[] split = command.split(" ", 2);
```
Would a plural name be a better name for an array?

kelvinou01 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,139 @@
+        String[] split = command.split(" ", 2);
+        String commandName = split[0];
+        int id = tasks.getSize();
+        if (split.length > 1){
```
Nothing major, but whitespace between `){` would be nice! May I suggest running `checkStyle` before submitting, it has made my life a lot easier!

kelvinou01 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,139 @@
+     * @throws DukeException If command cannot be parsed or is invalid.
+     */
+    public void parseFind(String command) throws DukeException {
+        String[] split = command.split(" ", 2);
```
Same as above

kelvinou01 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,146 @@
+ * Class that stores and manipulates tasks for Duke Bot.
+ */
+public class TaskList {
+    protected final ArrayList<Task> taskList;
```
Would a name like `tasks` be more concise?

kelvinou01 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,69 @@
+    /**
+     * Marks current Task as done.
+     */
+    public void done() {
```
Will a verb be more apt for this method name?

kelvinou01 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,69 @@
+    /**
+     * Marks current Task as not done.
+     */
+    public void undone() {
```
Same as above

kelvinou01 reviewed (commented):
Overall, the PR was pretty good, and adhered to the style guide! There were only some minor issues that I would like you to go over. Good job!

ErvinK123 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,66 @@
+                             + "| |_| | |_| |   <  __/\n"
+                             + "|____/ \\__,_|_|\\_\\___|\n";
+    private Scanner sc;
+    private boolean verbose;
```
Perhaps a more intuitive variable name here?

ErvinK123 commented on src/main/java/duke/task/EventTask.java:
```diff
@@ -0,0 +1,41 @@
+     */
+    public EventTask(String description) throws DateTimeParseException, DukeException {
+        super();
+        this.commandString = description;
```
Perhaps an overloaded constructor for superclass Task can be looked into

ErvinK123 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,79 @@
+ * Class that handles loading and saving of tasks for Duke Bot.
+ */
+public class Storage {
+    private FileWriter output;
```
Perhaps a more intuitive name could be used, that suggests the attribute is a filewriter

ErvinK123 reviewed (commented):
A few nits when naming variables, nothing major, should be good to merge when changed. 👍

fongyj committed:
Refactor for coding style

fongyj committed:
Add checkstyle and corrected style

fongyj committed:
Add GUI with javafx

fongyj committed:
Add handling of invalid index for mark, unmark and delete commands

fongyj committed:
Update creation of jar file

fongyj committed:
Merge branch 'branch-Level-10'

fongyj committed:
Add assertions

Included some assertions in Parser, Task and Storage class.

These assertions should help to flag out bugs in these classes during
runtime.

fongyj committed:
Improve code quality

Improved on quality of code according to code quality guide. Removed
instances of magic number and revised bad variable naming.

This makes the code easier to read.

fongyj committed:
Merge pull request #3 from fongyj/branch-A-Assertions

Add assertions

fongyj committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Storage.java

fongyj committed:
Merge pull request #4 from fongyj/branch-A-CodeQuality

Branch a code quality

fongyj committed:
Add C-DetectDuplicates functionality

fongyj committed:
Merge branch 'branch-BCD-Extension'

fongyj committed:
Optimised duplicate task checking

Used a HashSet to store task strings for duplicate checking.

Checking for duplicate via a HashSet is faster than checking for
duplicates by iterating through an ArrayList.

fongyj committed:
Implemented chick personality

fongyj committed:
Merge branch 'branch-A-Personality'

fongyj committed:
Add help command

fongyj committed:
Update javafx UI

fongyj committed:
Update javafx UI again

fongyj committed:
Correct style and add Ui image

fongyj committed:
Add README.md for Chick Bot

fongyj committed:
Update UI and personality

fongyj committed:
Update Ui.png

fongyj committed:
Update date parsing error and add close window on bye

fongyj committed:
Fix deadline and event task error mesesage

fongyj committed:
Rename Ui.png

fongyj committed:
Add Ui.png

fongyj committed:
Remove Ui.png

fongyj committed:
Fix task loading implementation issue

fongyj committed:
Remove unused image

fongyj committed:
Fix find command bug

Find command used to search using toString method of the tasks.
This will include (by:...) and (at:...) in the search. This will
mean that "find y" will return all deadlines and "find (" will
return all deadlines and events.

fongyj committed:
Fix storage and duplicate check implementation

Storage used to save only after bye command. Now save after every
modification to task list.
Also updated storage file chick.txt to not store uncheck command
as it is redundant.
Duplicate check was missing in the implementation of removing the
task from duplicate check after deleting the task from task list.

fongyj committed:
Fix storage and duplicate check implementation

Storage used to save only after bye command. Now save after every
modification to task list.
Also updated storage file chick.txt to not store uncheck command
as it is redundant.
Duplicate check was missing in the implementation of removing the
task from duplicate check after deleting the task from task list.

fongyj committed:
Merge branch 'master' of https://github.com/fongyj/ip

fongyj committed:
Fix user guide

Changed backslash in userguide to forward slash.
